### DISCUSSION TITLE
Error package path not exoported

### DISCUSSION THREAD

**Discussion by DarkTorias** [nextauthjs/next-auth#3467] [Help]
### Question 💬

Hi,
I just migrated to v4 and am getting an error after changing the nextauth api.
The error is as follows:

Error: Package subpath './providers/credentials' is not defined by "exports" in C:\folerProject\node_modules\next-auth\package.json

.next\server\pages\api\auth\[...nextauth].js (1:0) @ Object.next-auth/providers/credentials

> 1 | module.exports = require("next-auth/providers/credentials");

Thank you :)

### How to reproduce ☕️

Migrating to a v4

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by djordjeandjelkovic** [comment]
The same error here. Tried a lot of updates with versions, but nothing helped.

**Reply by pramade** [reply]
Same did you find any solution ?

**Comment by Mojtaba-NA** [comment]
i solved this problem by adding this line to **exports** field of the 'project/directoy/node_modules/next-auth/package.json' file, hope it helps.

**"./providers/credentials": "./providers/credentials.js"**
![Untitled](https://user-images.githubusercontent.com/45169963/147831509-bbfa0445-4de4-45db-90af-e8f52450ca11.jpg)

**Reply by balazsorban44** [reply]
As you can see in line 165, your added code should not be necessary, it's already exported. 🤷‍♂️

**Reply by Mojtaba-NA** [reply]
Yes, it should not but when i remove the line i wrote, the error shows up again.

**Reply by DarkTorias** [reply]
I do same as @Mojtaba-NikAeen and then I saw line 165, for that reason I wrote, because if i adding line, It working, but if I remove not, and It make no sense

**Comment by mattpardee** [comment]
I encountered this problem with Node 14.7. The problem goes away when using Node 16.13.1

**Comment by joshvr27** [comment]
anyone got any further with this? i could upgrade to 16 but unfortunately our production environment is node version 14.x only.

i can also see in next-auth v4.3.1 this is still an issue even with the glob pattern   "./providers/*": "./providers/*.js" in the next-auth package json.

**Comment by fanixk** [comment]
Also encountered this issue with node v12.19.0, but fixed using >= v12.22.7

**Comment by cristian495** [comment]
I was also struggling with this issue, the solution was change my Node version from 14.12.0 to 16.16.0

**Comment by gabeodame** [comment]
The solution is to update your node version. I had the same problem with node version 12 updated to latest and problem solved.

**Comment by AlAminKh03** [comment]
i'm using updated version of node.js but still can't encounter the error

**Comment by MohamadrezaKazemian** [comment]
it's 2023 and I'm still facing this error!

**Comment by bellobambo** [comment]
please i still face this issue

**Comment by bellobambo** [comment]
Found a solution🔥🔥

import NextAuth from 'next-auth'
import GithubProvider from "next-auth/providers/github"

export default NextAuth({
    providers: [
        GithubProvider({
            clientId: String(process.env.GITHUB_ID),
            clientSecret: String(process.env.GITHUB_SECRET),
        }),
    ],
})

**Reply by LokeshRatheee** [reply]
i dont want use github , i want use only provider

**Reply by ryanfavour4** [reply]
✅it works bro... for only github though. fact i hate next auth. firebase and auth0 thanks for existing

**Comment by freakaziod210** [comment]
Yeah. I am still encountering this problem.
I tried v4 with all of the node versions listed above and no dice.
Would have liked to use this for Auth but I guess I'll have to use firebase or Auth0

**Comment by tomaszs** [comment]
Still a bug

**Comment by luchillo17** [comment]
Why the hell was this moved to a discussion if it's still a bug?
~~I see this issue with `@auth/core@0.25.1`, can't update to the latest because `@auth/express` is behind in its dependency on core, so I get TS errors when adding the Mongo adapter.~~
I forced `ExpressAuth` and `MongoDBAdapter` to use `@auth/core@0.26.3` with npm `overrides`, still see this issue, ~~not sure if the issue is with `Credentials` specifically or with Express apps~~ it seems to be an issue specifically with Express because my other NextJS app is working fine...

**Comment by uroy98** [comment]
Hey, in your [...nextauth].js file try replacing the "import Providers from "next-auth/providers" with "import GoogleProvider from 'next-auth/providers/google'" because starting from version 4, next-auth has restructured how providers are imported.
